tiny_dnn/layers/average_pooling_layer.h

```diff
@@ -29,27 +29,28 @@ inline void tiny_average_pooling_kernel(
   float_t scale_factor,
   std::vector<typename partial_connected_layer::wi_connections> &out2wi) {
   for_i(parallelize, in_data[0]->size(), [&](size_t sample) {
-    const vec_t &in = (*in_data[0])[sample];
-    const vec_t &W  = (*in_data[1])[0];
-    const vec_t &b  = (*in_data[2])[0];
-    vec_t &out      = (*out_data[0])[sample];
+    const Tensor<> in((*in_data[0])[sample]);
```

Comment:
it's a workaround to get rid of the templated data type. @Randl new ideas about this ?
we should start thinking about how we define our `BaseTensor` based on dlpack tensor
https://github.com/dmlc/dlpack/blob/master/include/dlpack/dlpack.h